ci_tools/list_changed_files.py

```diff
@@ -0,0 +1,23 @@
+""" Script to list the python files changed by the PR
+"""
+import argparse
+
+parser = argparse.ArgumentParser(
+    description='Collect the files that have been modified by the PR to check their docstrings')
```

Comment:
Can you reuse `git_evaluation_tools.py` for this? That way we don't need to maintain 2 codes which do very similar things

Reply:
Thank you for pointing that out! i will refactor my code. I just wanna touch bases on the overall approach here. Do we concur on having the pull requester fix all the docstrings in the files he has modified? or should we limit it to the actual objects he has modified only. That way, instead of checking the files that have been changed i should rather check the lines and get solely the objects to whom these lines belong.

Reply:
For me the ideal approach would be to have the pull requester fix any docstrings he has modified and any errors that are new. That way if his changes cause example code elsewhere to become outdated he has to fix that too.

Some of the files in pyccel are really big so fixing everything seems a bit mean. In addition, we are likely to have lots of people changing docstrings at the same time as some files are worked on by everyone often.

Let me know if you want to discuss how to implement this

Reply:
> For me the ideal approach would be to have the pull requester fix any docstrings he has modified and any errors that are new. That way if his changes cause example code elsewhere to become outdated he has to fix that too.
The `sphinx` extension `doctest` will take care of this. I will create an action for it once `sphinx doc` is ready. It will make it very easy to detect the kind of errors you mentioned.
What do you think about the [new](https://github.com/pyccel/pyccel/pull/1292/commits/5d12c239a8632cd6770cc9de2d3f68ae31a43607) approach here. it only checks the objects that were modified, not the whole file. However, if a method is modified, it will also include its class docstring. The module docstring is always included. Please let me know if I understood your comment correctly.

Reply:
Ok we can do this later if you prefer :+1:

This seems ok for now but it doesn't really handle my comment. For example imagine that the interface of `Variable` or `Assign` is changed. The docstring of `FunctionDef` may now be wrong.

I was hoping that you could do something like:

- Get current problems in base branch as `base.json`
- Get current problems in new branch as `new.json`
- Report anything in `new.json` which isn't in `base.json`

Of course the exact solution will depend what the output of sphinx and co looks like. I don't know if there is a json option? Or something else that would be easy to manipulate?

Reply:
> This seems ok for now but it doesn't really handle my comment. For example imagine that the interface of Variable or >Assign is changed. The docstring of FunctionDef may now be wrong.

When you say that `FunctionDef` docstring may now be wrong. Do you mean the example/s present in it? or the actual description, parameters and return values? If it is the later, I do not know of a tool to check it.

Reply:
I mean the examples. I can't think of a reason why a change in the `Variable` interface would lead to a change in the arguments/results/definition of a `FunctionDef` without any code breaking

Reply:
That's why I said we can do this later if you haven't yet added the code that checks the examples

Reply:
Okay great!